include/deal.II/fe/fe.h

```diff
@@ -2991,6 +3008,72 @@ FiniteElement<dim,spacedim>::face_system_to_component_index (const unsigned int
 
 template <int dim, int spacedim>
 inline
+const FiniteElement<dim,spacedim> &
+FiniteElement<dim,spacedim>::get_child_element (const ComponentMask &mask) const
+{
+  const unsigned int n_total_components = this->n_components();
+  Assert(n_total_components == mask.size() || mask.size()==0,
+         ExcMessage("The given ComponentMask has the wrong size."));
+  Assert(mask.n_selected_components()>0,
+         ExcMessage("You need at least one selected component."));
+
+  const unsigned int n_selected = mask.n_selected_components(n_total_components);
+  const unsigned int first_selected = mask.first_selected_component(n_total_components);
+
+  // This offset keeps track how many components we have skipped so far:
+  int offset = 0;
+  // Current best candidate for the return value:
+  const FiniteElement<dim,spacedim> *result = this;
+
+  // keep selecting one of the base elements of an FESystem:
+  while (n_selected < result->n_components())
+    {
+      bool updated = false;
+      for (unsigned int base=0; base<result->n_base_elements(); ++base)
+        {
+          // how many components does this base element represent?
+          const unsigned int components_here
+            = result->element_multiplicity(base)*result->base_element(base).n_components();
+
+          if (first_selected >= offset
+              &&
+              first_selected+n_selected <= offset+components_here)
+            {
+              // the current base element is or contains what we are looking
+              // for:
+              Assert(result != &(result->base_element(base)),
+                     ExcMessage("You can not select part of a FiniteElement!"));
```

Comment:
I'm not sure I understand this error message. Are you saying that you can't subdivide something that's not divisible? Could this error ever happen if `base != 0`?